mailing_list/views.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.http import Http404
+from bulk_email.models import Optout
+from edxmako.shortcuts import render_to_response
+from opaque_keys.edx.keys import CourseKey
+
+from fun.utils import get_course
+
+def unsubscribe(request, course_id):
+    """
+    Create an optout resource related to the user retrieved through the email address
+    passed as argument and the current course.
+    """
+    course_key = CourseKey.from_string(course_id)
+    course = get_course(course_id)
+    email = request.GET.get('u')
+    user = None
+
+    if not course or not email:
+        raise Http404
+
+
+    try:
+        user = User.objects.get(email=email)
+    except User.DoesNotExist:
+        # Silently fail to not reveal existing account
```

Comment:
actually, if you don't want to reveal existing account, you should pretend that you un.subscribed the email as requested?

Reply:
oh ok that's what you did actually. So the comment is misleading:
```suggestion
        # Silently pass and display the page as if the un.subscribe succeeded in order to not reveal any existing account
```